cpu/fe310/include/irq_arch.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2017 Ken Rabold
+ *               2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup         cpu_fe310
+ * @{
+ *
+ * @file
+ * @brief           Implementation of the kernels irq interface
+ *
+ * @author          Ken Rabold
+ * @author          Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef IRQ_ARCH_H
+#define IRQ_ARCH_H
+
+#include <stdint.h>
+#include "irq.h"
+#include "cpu_conf.h"
+#include "cpu.h"
+
+#include "vendor/encoding.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern volatile int fe310_in_isr;
+
+/**
+ * @brief Enable all maskable interrupts
+ */
+static inline __attribute__((always_inline)) unsigned int irq_enable(void)
+{
+    /* Enable all interrupts */
+    set_csr(mstatus, MSTATUS_MIE);
+    return read_csr(mstatus);
```

Comment:
```suggestion
    unsigned retval = read_csr(mstatus);
    set_csr(mstatus, MSTATUS_MIE);
    return retval;
```

Care to sneak in a somewhat related bug fix? According to the doc, the IRQ state *before* interrupts were enabled should be returned. (The state after they were enabled should always be "enabled" in any case, so there is little point in returning it.)